server/routes/fetchThreadForSnapshot.ts

```diff
@@ -12,15 +12,15 @@ const fetchThreadForSnapshot = async (models: DB, req: Request, res: Response, n
   if (!snapshot) return next(new Error(Errors.InvalidSnapshot));
   if (!chain) return next(new Error(Errors.InvalidChain));
 
-  const thread = await models.OffchainThread.findOne({
+  const thread = await models.OffchainThread.findAll({
     where: { 
       chain: chain,
       snapshot_proposal: snapshot,
     }
   });
   if (!thread) return res.json({ status: 'Failure', message: '' });
```

Comment:
i believe the `findAll` above will return an array, even when empty. this check will never return failed!

the const `thread` should be renamed to `threads` and `threads.length < 1` could be the check.